autoload/airline/extensions.vim

```diff
@@ -138,7 +138,9 @@ function! airline#extensions#load()
     call airline#extensions#netrw#init(s:ext)
   endif
 
-  call airline#extensions#ycm#init(s:ext)
+  if exists(':YcmDiag')
```

Comment:
We can't check for a command because YCM sets up them on the VimEnter event. Also since it's a pretty heavy plugin a lot of people (me included) load it only on demand. I think it would be best to check for a variable here, disabled by default, that the user can use to load the extension or not.